runhouse/resources/serverless/aws_lambda_function.py

```diff
@@ -784,47 +772,76 @@ def aws_lambda_function(
     ):
         # Try reloading existing function
         return AWSLambdaFunction.from_name(name=name)
 
-    # ------- arguments validation -------
-    # TODO: use error msg inside the exception
-    if paths_to_code is None or len(paths_to_code) == 0:
-        logger.error("Please provide a path to the lambda handler file.")
-        raise RuntimeError
-    if handler_function_name is None or len(handler_function_name) == 0:
-        logger.error(
-            "Please provide the name of the function that should be executed by the lambda."
-        )
-        raise RuntimeError
-    if runtime is None or runtime not in SUPPORTED_RUNTIMES:
-        logger.error(
-            f"Please provide a supported lambda runtime, should be one of the following: {SUPPORTED_RUNTIMES}"
-        )
-        raise RuntimeError
-    if args_names is None:
-        logger.error(
-            "Please provide the names of the arguments provided to handler function, in the order they are"
-            + " passed to the lambda function."
-        )
-        raise RuntimeError
-
     # TODO: [SB] in the next phase, maybe add the option to create func from git.
 
+    if env is not None:
+        if isinstance(env, Env):
+            reqs = env.reqs
+            env_vars = env.env_vars
+            if not isinstance(env_vars, dict) and (
+                len(env_vars) > 0 or env_vars is not None
+            ):
+                raise Exception("AWS Lambda accepts env_vars of dict type only")
+        elif isinstance(env, dict):
+            reqs = env["reqs"]
+            env_vars = env["env_vars"]
+        else:
+            raise Exception(
+                "Env's type could be runhouse Env or dictionary. Please provide env in the correct type"
+                + " and rerun."
+            )
+        if isinstance(reqs, str):
+            reqs = AWSLambdaFunction._reqs_to_list(reqs)
+        if isinstance(reqs, list):
+            reqs = [req for req in reqs if req != "./"]
+    else:
+        reqs, env_vars = None, None
+
     fn_pointers = (
-        Function._extract_pointers(fn, reqs=env or []) if callable(fn) else None
+        Function._extract_pointers(fn, reqs=reqs or []) if callable(fn) else None
     )
 
+    if fn is None:
+        # ------- arguments validation -------
+        if paths_to_code is None or len(paths_to_code) == 0:
+            raise RuntimeError("Please provide a path to the lambda handler file.")
+        if handler_function_name is None or len(handler_function_name) == 0:
+            raise RuntimeError(
+                "Please provide the name of the function that should be executed by the lambda."
+            )
+        if runtime is not None and runtime not in SUPPORTED_RUNTIMES:
+            raise RuntimeError(
+                "Please provide a supported lambda runtime, should be one of the"
+                + f" following: {SUPPORTED_RUNTIMES}"
+            )
+        if args_names is None:
```

Comment:
can't we also extract these if the `fn` is passed in? seems like it will always be passed into the class with `None`

Reply:
If the fn is provided, we are indeed extracting the arguments names. 
But if fn is not provided, we expect the user to provide the args names. If the arguments will not be passed, I've added a piece of code which parses the hander and extracts the arguments names.